build/build-bundle.js

```diff
@@ -29,26 +29,13 @@ const buildBundle = async () => {
     for (const targetOpts of TARGET_ENTRIES) {
       const mergedOpts = { ...baseOptions, ...targetOpts };
 
-      if (IS_WATCH_MODE) {
-        mergedOpts.watch = {
-          onRebuild(error) {
-            if (error)
-              console.error(
-                `[ESBuild Watch] (${targetOpts.entryPoints[0]}) Failed to rebuild bundle`
-              );
-            else
-              console.log(
-                `[ESBuild Watch] (${targetOpts.entryPoints[0]}) Sucessfully rebuilt bundle`
-              );
-          },
-        };
-      }
+      const context = await esbuild.context(mergedOpts);
 
-      const { errors } = await esbuild.build(mergedOpts);
-
-      if (errors.length) {
-        console.error(`[ESBuild] Bundle failed with ${errors.length} errors`);
-        process.exit(1);
+      if (IS_WATCH_MODE) {
+        await context.watch();
+      } else {
+        await context.rebuild();
```

Comment:
I'm not entirely sure if esbuild logs errors internally, even though this does happen when using watch mode. Maybe it would be safe to still add a check like we had before.